### PR TITLE
Restructure package

### LINKED ISSUE #335: Avoid any output of rules to be placed into data folder
## Describe the feature you'd like to see

I think it is a good idea to move all processed outputs of rules in folders different than "data", to keep in data only the raw files.
The idea is to simplify the organization and the "cleaning" process in case we would like to do delete all intermediate files: with the proposed change, the user shall only delete resources/networks/results folders and keep data as-is.
Currently base_network, download/clean_osm_data do save output csv files into the data folder.

### LINKED ISSUE #354: Restructure folder package to ease readability, usage and manutentibility
## Describe the feature you'd like to see
The package shall be restructured so to store data in folders in an organized way, i.e. put in data all raw data and organize output files in subfolders to ease the usage.

### PR DESCRIPTION
**Pull request by davide-f**
# Closes #354 and closes #335

## Changes proposed in this Pull Request

This PR aims at providing a more structured filestructure of the package to:
- Ease the readibility
- Simplify the debugging phase (e.g. to reset the repo without deleting raw files, just delete resources/ networks/ and results/ folders
- Ease the integration with the PyPSA-Earth-Sec package
- Fix google drive downloader interface ([PR also on googledrive downloader](https://github.com/ndrplz/google-drive-downloader/pull/28) to fix a bug there)

Note: the zenodo package of the new data tutorial file is not loaded yet.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by davide-f** [comment]
@Hazem-IEG This PR may be of interest for you.
This is a proposal on how to restructure the pypsa-earth model that may be in handy for the -sec model.
Feel free to add comments

**Comment by davide-f** [comment]
@all this PR may stay on hold until most parallel PRs are not complete; especially #348 and #343 , yet I think it's a nice-to-have to finalize the v0.1

**Review by pz-max — COMMENTED** [review]
Looks good.
1. Would only merge the main into the branch
2. Move `ssp` from `pypsa-africa/resources` to `pypsa-africa/data` such that no download is triggered when deleting resource folder. This will require some debugging in the load part in the `add_electricity.py` and maybe some adjustment in the config.yaml

Happy to create a PR on that. Let me know :)

**Comment by davide-f** [comment]
> Looks good.
> 
> 1. Would only merge the main into the branch
> 2. Move `ssp` from `pypsa-africa/resources` to `pypsa-africa/data` such that no download is triggered when deleting resource folder. This will require some debugging in the load part in the `add_electricity.py` and maybe some adjustment in the config.yaml
> 
> Happy to create a PR on that. Let me know :)

That was on my local pr to test, commits now pushed :)

**Comment by davide-f** [comment]
@pz-max now it should be ready, when the CI ends

File: scripts/retrieve_databundle_light.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def download_and_unzip_gdrive(config, rootpath, hot_run=True, disable_progress=F
         gdd.download_file_from_google_drive(
             file_id=file_id,
             dest_path=file_path,
-            showsize=True,
+            showsize=not disable_progress,
```
Comment (pz-max) [inline scripts/retrieve_databundle_light.py:217]: looks weird. is this the correct syntax?

**Review by pz-max — APPROVED** [review]

**Comment by pz-max** [comment]
Great contribution :1st_place_medal:

File: scripts/retrieve_databundle_light.py (hunk 1 above)
Comment (pz-max) [reply scripts/retrieve_databundle_light.py:217]: Davide confirmed. It is weird and it's their syntax.

**Pull request merged**